crates/sui-protocol-config/src/lib.rs

```diff
@@ -454,6 +524,17 @@ impl ProtocolConfig {
         // To change the values here you must create a new protocol version with the new values!
         match version.0 {
             1 => Self {
+                max_tx_in_batch: Some(10),
+                max_modules_in_publish: Some(8),
+                max_arguments: Some(16),
+                max_type_arguments: Some(128),
+                max_type_argument_depth: Some(16),
+                max_pure_argument_size: Some(10 * 1024),
+                max_object_vec_argument_size: Some(128),
+                max_coins: Some(1024),
+                max_pay_recipients: Some(1024),
+                max_programmable_tx_commands: Some(16),
```

Comment:
64 unless someone wants it bigger

Reply:
Similar problem here, a max size would be 64 programmable tx commands * 16 call arguments * 10k max pure arg size would be 10MB, too big.

Reply:
same answer as above :) I do not think we should have the total limit on the size of a user transaction being based over a composition of those limits. It will be a nightmare in my opinion.
I would stick with a max user transaction size of 64K and stop it that way.
Would that help?
in other words the tension here is between valid scenarios that can use a number of arguments and still be reasonably small and less the 64K.
In any case if others (particularly @tnowacki are happy with this I am fine too). Though again let's have a limit of 64K for users transactions for now. Or a similar relatively low number (128K or whatever).

Reply:
I'm all for an overall `Transaction` size limit. The question is, how do we define the "size" of a transaction and how do we measure it deterministically? We could use `mysten-util-mem`, but if we have `usize` fields or pointers anywhere in it, then the result will vary depending on what hardware the validator is running. We could use the serialized size of the input protobuf, but that may not be the same as the bincode serialized size used by anemo. And anyway, we have talked about switching the tx input interface away from gRPC/protobuf.

I think we just need to pick an answer here, even if it's not perfect. I'd love opinions from any of you more rust-adept folks @dariorussi @mystenmark @sblackshear

Reply:
just a note that we don't use protobuf anywhere. our grpc interfaces use bincode as well. So I think it should be quite simple and deterministic to just enforce the limit on the serialized payload.

Reply:
Ah right I forgot that detail. We do use protobuf for the narwhal gRPC interface, so my brain was subconsciously thinking we also did for the sui validator one, even though I knew we didn't 🤦